### PR TITLE
[DOCS] Added compatibility section in doc + updated release notes

### PR DESCRIPTION
**Pull request by ezimuel**
This PR provides the following changes in the documentation:

- added a compatibility section;
- updated the release notes up to 7.14;

Moreover it updates the README with the new compatibility section content.

### CODE REVIEW

File: docs/compatibility.asciidoc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[[compatibility]]
+== Compatibility
+
+Language clients are forward compatible; meaning that clients support communicating
+with greater or equal minor versions of Elasticsearch. Elasticsearch language
+clients are only backwards compatible with default distributions and without
+guarantees made.
```
Comment (szabosteve) [inline docs/compatibility.asciidoc:7]: In the rest of the Client books, we added the compatibility information to the `Installation` section as a level-three sub-section:
```
[discrete]
[[compatibility]]
=== Compatibility
```
If there is no particular reason to add this info as a separate section, I would suggest moving this content under the Installation section.

File: docs/index.asciidoc [hunk 2]
```diff
@@ -5,6 +5,8 @@ include::{asciidoc-dir}/../../shared/attributes.asciidoc[]
 
 include::overview.asciidoc[]
 
+include::compatibility.asciidoc[]
```
Comment (szabosteve) [inline docs/index.asciidoc:8]: If you move the compatibility info under `Installation`, then this item can be deleted.
```suggestion
```

**Review by szabosteve — COMMENTED** [review]
One suggestion about the place of the compatibility info inside the PHP book.

**Review by ezimuel — COMMENTED** [review]
I just moved the `compatibility` section into `installation` as suggested.

**Review by szabosteve — APPROVED** [review]
LGTM!

**Pull request merged**